pkg/tcl/apitcl/v1/testworkflowmetrics.go

```diff
@@ -166,61 +98,206 @@ func (s *apiTCL) sendRunWorkflowTelemetry(ctx context.Context, workflow *testwor
-			isKubeshopGitURI = true
+	for _, step := range spec.Setup {
+		if hasStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	for _, step := range spec.After {
+		if hasStepKubeshopGitURI(step) {
+			return true
 		}
 	}
 
-	out, err := telemetry.SendRunWorkflowEvent("testkube_api_run_test_workflow", telemetry.RunWorkflowParams{
-		RunParams: telemetry.RunParams{
-			AppVersion: version.Version,
-			DataSource: dataSource,
-			Host:       host,
-			ClusterID:  clusterID,
-		},
-		WorkflowParams: telemetry.WorkflowParams{
-			TestWorkflowSteps:          int32(len(workflow.Spec.Steps)),
-			TestWorkflowImage:          image,
-			TestWorkflowArtifactUsed:   hasArtifacts,
-			TestWorkflowKubeshopGitURI: isKubeshopGitURI,
-		},
-	})
+	return false
+}
+
+// hasTemplateKubeshopGitURI checks if the test workflow spec has a git URI that contains "kubeshop"
+func hasTemplateKubeshopGitURI(spec testworkflowsv1.TestWorkflowTemplateSpec) bool {
+	if isKubeshopGitURI(spec.Content) {
+		return true
+	}
+
+	for _, step := range spec.Steps {
+		if hasTemplateStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	for _, step := range spec.Setup {
+		if hasTemplateStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	for _, step := range spec.After {
+		if hasTemplateStepKubeshopGitURI(step) {
+			return true
+		}
+	}
 
+	return false
+}
+
+// hasTemplateStepKubeshopGitURI checks if the step has a git URI that contains "kubeshop"
+func hasTemplateStepKubeshopGitURI(step testworkflowsv1.IndependentStep) bool {
+	for _, step := range step.Setup {
+		if isKubeshopGitURI(step.Content) {
+			return true
+		}
+		if hasTemplateStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	for _, step := range step.Steps {
+		if isKubeshopGitURI(step.Content) {
+			return true
+		}
+		if hasTemplateStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	return false
+}
+
+// hasStepKubeshopGitURI checks if the step has a git URI that contains "kubeshop"
+func hasStepKubeshopGitURI(step testworkflowsv1.Step) bool {
+	for _, step := range step.Setup {
+		if isKubeshopGitURI(step.Content) {
+			return true
+		}
+		if hasStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	for _, step := range step.Steps {
+		if isKubeshopGitURI(step.Content) {
+			return true
+		}
+		if hasStepKubeshopGitURI(step) {
+			return true
+		}
+	}
+	return false
+}
+
+// isKubeshopGitURI checks if the content has a git URI that contains "kubeshop"
+func isKubeshopGitURI(content *testworkflowsv1.Content) bool {
+	switch {
+	case content == nil:
+		return false
+	case content.Git == nil:
+		return false
+	case strings.Contains(content.Git.Uri, "kubeshop"):
+		return true
+	default:
+		return false
+	}
+}
+
+// getDataSource returns the data source of the content
+func getDataSource(content *testworkflowsv1.Content) string {
+	var dataSource string
+	if content != nil {
+		if len(content.Files) != 0 {
+			dataSource = "files"
+		} else if content.Git != nil {
+			dataSource = "git"
+		}
```

Comment:
Actually both can be used at the same time. Probably worth to prioritize Git then